src/coefficients/stdfunction_coefficient.hpp

```diff
@@ -67,24 +94,71 @@ class StdFunctionVectorCoefficient : public mfem::VectorCoefficient {
    * void(mfem::Vector &, mfem::Vector &). The first argument of the function is
    * the position, and the second argument is the output of the function.
    */
+
+  /**
+   * @brief Constructor that takes in a time-independent function of space that returns a vector
+   *
+   * @param[in] dim: number of components in the output vector
+   * @param[in] func: a function of space and time that returns a double
+   */
   StdFunctionVectorCoefficient(int dim, std::function<void(mfem::Vector&, mfem::Vector&)> func);
 
   /**
-   * @brief Evalate the coefficient at a quadrature point
+   * @brief Constructor that takes in a time-dependent function of space that returns a double
+   *
+   * @param[in] dim: number of components in the output vector
+   * @param[in] func: a function of space and time that returns a double
+   */
+  StdFunctionVectorCoefficient(int dim, std::function<void(mfem::Vector&, mfem::Vector&, double)> func);
+
+  /**
+   * @brief Evaluate the coefficient at a quadrature point
    *
    * @param[out] V The evaluated coefficient vector at the quadrature point
    * @param[in] T The element transformation for the evaluation
    * @param[in] ip The integration point for the evaluation
    */
   virtual void Eval(mfem::Vector& V, mfem::ElementTransformation& T, const mfem::IntegrationPoint& ip);
+  
+  bool is_time_dependent() const { return is_time_dependent_; } 
+
+  friend StdFunctionVectorCoefficient d_dt(const StdFunctionVectorCoefficient &, const double);
+  friend StdFunctionVectorCoefficient d2_dt2(const StdFunctionVectorCoefficient &, const double);
 
 private:
   /**
    * @brief The function to evaluate for the coefficient
    */
-  std::function<void(mfem::Vector&, mfem::Vector&)> func_;
+  std::function<void(mfem::Vector&, mfem::Vector&, double)> func_;
+  bool is_time_dependent_;
 };
 
+inline StdFunctionVectorCoefficient d_dt(const StdFunctionVectorCoefficient & y, const double dt = 1.0e-8) {
+  return StdFunctionVectorCoefficient(y.vdim, [dt, y = y.func_](mfem::Vector& x, mfem::Vector & return_value, double t) {
```

Comment:
I'm probably missing something here, but why is dt defaulted to 1.0e-8?

Reply:
it was based on the assumption that the user is picking appropriate units to get the characteristic time on the order of unity,

![first_derivative_error](https://user-images.githubusercontent.com/16393433/94727181-2f6f0900-0313-11eb-95b6-71f39e08da18.png)

Reply:
I don't think that's a good assumption.  People generally use physical units (seconds, microseconds) and don't rescale.